src/main/java/io/cdap/plugin/snowflake/common/client/SnowflakeAccessor.java

```diff
@@ -143,6 +150,12 @@ public void checkConnection() {
       connection.getMetaData();
     } catch (SQLException e) {
       throw new ConnectionTimeoutException("Cannot create Snowflake connection.", e);
+    } catch (NullPointerException e) {
+      String errorMessage = String.format("Unable to create Snowflake connection due to missing Username or password." +
+        "Failed reason is %s .", e.getMessage());
+      String errorReason = "Cannot create Snowflake connection. Username or password is missing.";
+      throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+        errorReason, errorMessage, ErrorType.SYSTEM, true, e);
```

Comment:
similar question here: if the user & password is provided by cx, why is it `SYSTEM` error?

Reply:
Also do we not have validations for these configs in `configurePipeline` & `prepareRun`?